File: JavaScript/Behavioral/Observer.js
/* Observer is a behavioral design pattern that lets 
you define a subscription mechanism to notify multiple 
objects about any events that happen to the object they’re observing. */

// creating the Observer class and the Observable
class Observer {
    update() {
        console.log('There was an update!!!');
    }
}

class Observable {
    observers = [];

    notifyObservers() {
        if (this.observers.length != 0) {
            for(let i = 0; i < this.observers.length ; i++) {
                this.observers[i].update();
            }
        }
    }
}

// instantiating some observers and updating
let observable = new Observable();
observable.observers.push(new Observer(), new Observer(), new Observer());
observable.notifyObservers();

File: CSharp/Behavioral/Observer.cs
namespace Behavioral;

/* Observer is a behavioral design pattern that lets
you define a subscription mechanism to notify multiple
objects about any events that happen to the object they’re observing. */

// creating the Observer class and the Observable
public class Observer
{
    public class ObserverClass
    {
        public void Update()
        {
            Console.WriteLine("There was an update!!!");
        }
    }

    public class Observable
    {
        public List<ObserverClass> Observers { get; set; } = new List<ObserverClass>();

        public void NotifyObservers()
        {
            if (Observers.Count is 0) return;
            foreach(var observer in Observers)
            {
                observer.Update();
            }
        }
    }
}